### PR TITLE
* reduce severity of mask mismatch

### PR DESCRIPTION
**Pull request by shadeofblue**
* refactor `taskkeeper.check_support` tests :p

closes: https://github.com/golemfactory/golem/issues/3438

### CODE REVIEW

File: tests/golem/task/test_taskkeeper.py [hunk 1]
```diff
@@ -38,54 +38,89 @@ def async_run(request, success=None, error=None):
...
+        self.assertIn(UnsupportReason.MASK_MISMATCH, supported.desc)
+
+
+class TestTaskHeaderKeeper(LogTestCase):
+    def test_init(self):
+        tk = TaskHeaderKeeper(
+            environments_manager=EnvironmentsManager(),
+            node=dt_p2p_factory.Node(),
+            min_price=10.0)
+        self.assertIsInstance(tk, TaskHeaderKeeper)
```
Comment (jiivan) [inline tests/golem/task/test_taskkeeper.py]: 8-|

**Review by jiivan — APPROVED** [review]

File: tests/golem/task/test_taskkeeper.py (hunk 1 above)
Comment (shadeofblue) [reply tests/golem/task/test_taskkeeper.py]: ugh ... that's not my code ;)

File: tests/golem/task/test_taskkeeper.py [hunk 2]
```diff
@@ -38,54 +38,89 @@ def async_run(request, success=None, error=None):
...
+
+        with self.assertNoLogs('golem.task.taskkeeper', level='INFO'):
+            supported = self.tk.check_support(header)
+
+        self.assertFalse(supported)
+        self.assertIn(UnsupportReason.MASK_MISMATCH, supported.desc)
+
+
+class TestTaskHeaderKeeper(LogTestCase):
+    def test_init(self):
```
Comment (mfranciszkiewicz) [inline tests/golem/task/test_taskkeeper.py:118]: This test is redundant :)

File: golem/environments/environment.py [hunk 3]
```diff
@@ -37,6 +37,13 @@ def ok(cls) -> 'SupportStatus':
     def err(cls, desc) -> 'SupportStatus':
         return cls(False, desc)
 
+    @property
+    def err_reason(self):
+        try:
+            return list(self.desc.keys())[0]
```
Comment (mfranciszkiewicz) [inline golem/environments/environment.py:43]:
```suggestion
            return next(iter(self.desc))
```
You'd need to catch a `StopIteration` exception though.

**Review by mfranciszkiewicz — APPROVED** [review]

File: tests/golem/task/test_taskkeeper.py (hunk 2 above)
Comment (shadeofblue) [reply tests/golem/task/test_taskkeeper.py:118]: repeat after me: it's not my code ...

**Pull request merged**